### PR TITLE
Patient UI with Getting and posting data

### PR DESCRIPTION
**Pull request by pmadria**
This creates a patient UI form that is able to take and submit data. Run `npm start`to start the react server for the form and run `bundle exec ruby app.rb` to obtain information from the form in a get or post data about the patient in another server.

Localhost:3000 => react server
Localhost:8000 => localhost server for patient 

Sample GET request: http://localhost:8000/Patient/ for Postman or whatever tool you use.

### CODE REVIEW

File: frontend/src/PatientForm.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        <input
+          type="text"
+          value={email}
+          onChange={(event) => setEmail(event.target.value)}
+        />
+      </label>
+      <label>
+        Date of Birth:
+        <input
+          type="text"
```
Comment (Jammjammjamm) [inline frontend/src/PatientForm.js:109]: I think you want `date` rather than `text` here.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+require 'sinatra'
+require 'json'
+
+set :port, 8000
+set :static, true
+set :public_folder, 'static'
+set :views, 'frontend/src'
+
+post '/Patient/' do
```
Comment (Jammjammjamm) [inline app.rb:11]:
```suggestion
post '/Patient' do
```

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  @@mi = params['name'][0]['given'][1]
+  @@ln = params['name'][0]['family']
+  @@suffix = params['name'][0]['suffix']
+  @@gender = params['gender']
+  @@telecom = params['telecom'][0]['phone']
+  @@email = params['telecom'][0]['email']
+  @@birth_date = params['birthDate']
+end
+
+get '/Patient/' do
```
Comment (Jammjammjamm) [inline app.rb:25]:
```suggestion
get '/Patient' do
```

**Review by Jammjammjamm — COMMENTED** [review]
[Foreman](https://github.com/ddollar/foreman) can handle this situation where you need to run ruby & js:

- `gem install foreman`
- create a file called `Procfile.dev` that looks something like:
```
api: bundle exec ruby app.rb
client: npm start
```
- run `bundle exec foreman start -f Procfile.dev` (or better yet, make a `bin/serve` script to do this)


Is this what it is supposed to look like?
Screen Shot 2021-03-04 at 12 16 29 PM

**Comment by ssayer** [comment]
Closing this due to the move to Rails, but leaving the branch in case we want to refer to the JS code later

**Pull request closed without merging**